src/py/pyodide/_package_loader.py

```diff
@@ -299,6 +306,31 @@ def get_dynlibs(archive: IO[bytes], suffix: str, target_dir: Path) -> list[str]:
     ]
 
 
+def get_dist_source(dist: Distribution) -> str | None:
```

Comment:
Could you add a docstring with at least 1 sentence summary indicating what exactly is the source. I gather it's the package name (but that, requires reading the code). Also a short summary of the logic here would be helpful.

Also shouldn't this be at somewhat shared with micropip: I imagine the mechanism for parsing importlib distributions is similar.

Reply:
It's used in `loadPackage` messages:
```js
messageCallback(`${pkg} already loaded from ${loaded}`);
```
https://github.com/pyodide/pyodide/blob/main/src/js/load-package.ts#L369
so it's primarily to explain to the user why we think the package has already been loaded. It's important that there be some entry in `loadedPackages` to prevent `loadPackage` from copying another version of the package over it.